### PR TITLE
chore(networking): Upgrade `trust-dns`

### LINKED ISSUE #2464: Upgrade `trust-dns` to fix `RUSTSEC-2020-0014`
We should upgrade `trust-dns` to allow us to upgrade our version of rusqulite to avoid this vuln.

### PR DESCRIPTION
**Pull request by LucioFranco**
This upgrades `trust-dns` and allows us to fix the RUSTSEC issue that came up earlier this week.

Currently, blocked by https://github.com/bluejekyll/trust-dns/pull/1087 and #2511 is open to upgrade to the crates version.

Closes #2464

Signed-off-by: Lucio Franco <luciofranco14@gmail.com>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -419,3 +418,9 @@ harness = false
 
 [patch.'https://github.com/tower-rs/tower']
 tower-layer = "0.3"
+
+[patch.crates-io]
+trust-dns-server = { git = "https://github.com/bluejekyll/trust-dns" }
```
[the remark this answers is not in the document]
Comment (LucioFranco) [reply Cargo.toml:423]: https://github.com/timberio/vector/issues/2511

**Comment by Hoverbear** [comment]
@LucioFranco The CI is unhappy with this PR. 

Otherwise it looks good :)

**Comment by LucioFranco** [comment]
I can't seem to reproduce any of these failures locally but they seem to be very consistent on CI :/

File: deny.toml [hunk 2]
```diff
@@ -1,7 +1,5 @@
 [advisories]
 ignore = [
-    "RUSTSEC-2020-0014", # Advisory for dev dependency of `trust-dns` for testing
-                         # dns servers.
     "RUSTSEC-2020-0002", # tracked in #1639
     "RUSTSEC-2020-0001", # dev dependency only
```
Comment (fanatid) [inline deny.toml:4]: `RUSTSEC-2020-0001` not required with this changes too?
https://rustsec.org/advisories/RUSTSEC-2020-0001.html

Comment (LucioFranco) [reply deny.toml:4]: Yup, good catch we can remove that one as well!

**Comment by LucioFranco** [comment]
https://github.com/timberio/vector/issues/2635#issue-620514170

**Pull request closed without merging**